### ISSUE TITLE
rio.write_nodata returns ValueError: cannot convert float NaN to integer

### ISSUE THREAD

**Issue by Jack-Hayes** [egagli/easysnowdata#7]
### Environment Information

-   easysnowdata version: v0.0.8
-   Python version: 3.12.4
-   Operating System: Windows 11

### Description

After loading in esa_worldcover data as seen in the documentation, I wanted to check id there were NaN values with "np.isnan(esa_worldcover.rio.nodata)". Running this returns "ValueError: cannot convert float NaN to integer" on the first run, but when you run the same np.isnan(esa_worldcover.rio.nodata)" for a second time, expected results are returned. 

Found that this was an issue with remote_sensing.py line 254:
worldcover_da = worldcover_da.rio.write_nodata(0, encoded=True)

encoded=True seems to be the cause of this.

### What I Did

```
Running "esa_worldcover.rio.write_nodata(0, encoded=False, inplace=True)" after you load in esa_worldcover solces this issue
```

**Comment by egagli** [comment]
Thanks for finding this! Will push up a fix soon

**Comment by egagli** [comment]
issue fixed by commit b624a19141913db05410b4645c0830ba1c16f44f, newest release works!
